### DISCUSSION TITLE
samples: subsys: usb: mass: Use &flash0 storage_partition for USB mass storage

### DISCUSSION THREAD

**Discussion by joaodullius** [zephyrproject-rtos/zephyr#37764] [Ideas] [answered]
What would be the necessary changes for samples/subsys/usb/mass to use internal &flash0 storage_partition for the mouting the USB device partition?

From the Kconfig, it appears that it only supports RAM, NAND and SD.

Most boards (at least the nrfxxxxDKs) now have a storage_partition in the end of the &flash. I with to mount the USB there:

      storage_partition: partition@3a000 {
	      label = "storage";
	      reg = <0x0003a000 0x00006000>;

**Comment by de-nordic** [comment]
You would need to modify the flash disk driver here: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/disk/flashdisk.c
to use DTS instead of taking info from Kconfig, or make DTS alternative source of flash disk configuration.

> From the Kconfig, it appears that it only supports RAM, NAND and SD.

These are only labels that the ELM FAT driver recognizes as mount points when doing path lookup; you can mount SD at NAND and SD at RAM, it does not matter (https://github.com/zephyrproject-rtos/fatfs/blob/94fcd6bfb3801ac0a5e12ea2f52187e0a688b90e/include/ffconf.h#L204)

**Reply by joaodullius** [reply]
@de-nordic,  can't I somehow point CONFIG_DISK_FLASH_DEV_NAME to the &flash0?

**Comment by de-nordic** [comment, marked as answer]
What is flash0? The `CONFIG_DISK_FLASH_DEV_NAME` points to flash control device; Do you mean that you want to point `CONFIG_DISK_FLASH_DEV_NAME` to a partition label?

**Reply by joaodullius** [reply]
Yes, to point to a flash partition on the internal flash of the device.
I'm targeting nrf52833dk_nrf52820 that has the following flash definition in the dts:

    &flash0 {
    
	    partitions {
		    compatible = "fixed-partitions";
		    #address-cells = <1>;
		    #size-cells = <1>;
    
		    boot_partition: partition@0 {
			    label = "mcuboot";
			    reg = <0x00000000 0xC000>;
		    };
		    slot0_partition: partition@C000 {
			    label = "image-0";
			    reg = <0x0000C000 0x17000>;
		    };
		    slot1_partition: partition@23000 {
			    label = "image-1";
			    reg = <0x00023000 0x17000>;
		    };
		    storage_partition: partition@3a000 {
			    label = "storage";
			    reg = <0x0003a000 0x00006000>;
		    };
	    };
    };

I was wondering if I could somehow from the Kconfig point to the storage_partition instead of the external flash.

**Reply by de-nordic** [reply]
Unfortunately not with current implementation, changes to the https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/disk/flashdisk.c and Kconfig are required.
Problem with Kconfig is that it can only take int, string or bool values, which means that passing DTS label/alias like `flash0` would be visible as "flash0" and the DTS macros can not deal with strings to get values for partition size and offset.
